temporian/beam/typing.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple, Union, Optional, List, Any
+
+import numpy as np
+import apache_beam as beam
+from temporian.core.typing import IndexKeyItem
+
+# When applicable, follow the same naming convention as temporian/core/typing.py
+#
+# The Beam computation relies eavily on tuples instead of dataclasses or named
+# tuples as it seems tuples are the most efficient solution.
+#
+# Individual features of a same event set are stored separately.
+# An important implication is that timestamps are effectively repeated for
+# each feature.
+
+# Temporian index in Beam.
+BeamIndexKeyItem = Union[int, bytes]
+BeamIndexKey = Tuple[BeamIndexKeyItem, ...]
```

Comment:
These two are the same as `NormalizedIndexKeyItem` and `NormalizedIndexKey`, does it make sense to have them separate?

Reply:
Yes, those are the same.
* Replaced.